user-interface/src/main/java/life/qbic/datamanager/views/general/contact/AutocompleteContactField.java

```diff
@@ -83,13 +86,21 @@ public AutocompleteContactField(String label) {
         .bind(Contact::getEmail,
             Contact::setEmail);
 
+    Div preselectLayout = new Div(contactSelection, selfSelect);
+    preselectLayout.addClassName("prefill-input-fields");
+
     Div layout = new Div(nameField, emailField);
     layout.addClassName("input-fields");
-    add(contactSelection, layout);
+
+    add(preselectLayout, layout);
     setItems(new ArrayList<>());
     clear();
   }
 
+  private void onSelfSelected(ClickEvent<Button> buttonClickEvent) {
+    setContact(userAsContact);
```

Comment:
You could go further and query the user information only here when it is needed 🤷 
No strong opinion.

Reply:
I like it!